### PR TITLE
Fix/108/facetVariableDetails should always be an array

### LINKED ISSUE #108: Backend: data[number].facetVariableDetails is not always an array
The Data API specifies always an array
```
mosaic.data[].facetVariableDetails | array | Max. items: 2
```
Yet when I make this GEMS1 request it's just a single object not in an array:

POST
```
{
	"config": {
		"facetVariable": [
			{
				"entityId": "EUPATH_0035127",
				"variableId": "ENVO_00000009"
			}
		],
		"outputEntityId": "EUPATH_0000096",
		"xAxisVariable": {
			"entityId": "EUPATH_0000096",
			"variableId": "EUPATH_0010367"
		},
		"yAxisVariable": {
			"entityId": "EUPATH_0000096",
			"variableId": "EUPATH_0011048"
		}
	},
	"filters": [
		{
			"entityId": "EUPATH_0035127",
			"stringSet": [
				"India"
			],
			"type": "stringSet",
			"variableId": "ENVO_00000009"
		}
	],
	"studyId": "GEMSCC0003-1"
}
``` 
Response (part of)
```
{
	"facetVariableDetails": {
		"variableId": "ENVO_00000009",
		"entityId": "EUPATH_0035127",
		"value": "India"
	}
}
```

The troublemaking facet variable is Country.

When I use `Case or control participant` the response is correctly array-like:
```
{
	"facetVariableDetails": [
		{
			"variableId": "EUPATH_0010375",
			"entityId": "EUPATH_0000096",
			"value": "Case"
		}
	]
}
```

Sorry I haven't tried more variables/entities to figure out more of an explanation.

### PR DESCRIPTION
**Pull request by d-callan**
resolves #108

### CODE REVIEW

File: R/utils-json.R [hunk 1]
```diff
@@ -72,6 +76,7 @@ getJSON <- function(.pd, evilMode) {
   namedAttrList <- getPDAttributes(.pd)
   class <- attr(.pd, 'class')[1] 
 
+  # why do we care about evilMode here??
```
Comment (asizemore) [inline R/utils-json.R:79]: We dont want to return stats when evilMode=T, so i assume this check is a second round of defense (the first round is in the class constructor function i believe)

File: tests/testthat/test-mosaic.R [hunk 2]
```diff
@@ -357,6 +357,36 @@ test_that("mosaic.dt() returns an appropriately sized data.table", {
 })
 
 test_that("mosaic() returns appropriately formatted json", {
+  map <- data.frame('id' = c('entity.binB', 'entity.binA', 'entity.cat4'),
+                    'plotRef' = c('yAxisVariable', 'xAxisVariable', 'facetVariable1'),
+                    'dataType' = c('STRING', 'STRING', 'STRING'),
+                    'dataShape' = c('CATEGORICAL', 'CATEGORICAL', 'CATEGORICAL'), stringsAsFactors=FALSE)
+
+  df <- testDF[testDF$entity.cat4 == 'cat4_a',]
```
Comment (asizemore) [inline tests/testthat/test-mosaic.R:365]: testDF also has an `entity.cat1` column which only has 1 value

**Review by asizemore — APPROVED** [review]
Looks good! Tested with `mosaic` and checked the output json file.

File: tests/testthat/test-mosaic.R (hunk 2 above)
Comment (d-callan) [reply tests/testthat/test-mosaic.R:365]: very nice. didnt realize that. ill update.

**Pull request merged**